pkg/chunk/table_manager.go

```diff
@@ -39,11 +40,36 @@ var (
 		Name:      "dynamo_table_capacity_units",
 		Help:      "Per-table DynamoDB capacity, measured in DynamoDB capacity units.",
 	}, []string{"op", "table"})
+
+	expectedMinIndexTableNumber = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "cortex",
+		Name:      "expected_min_index_table_number",
+		Help:      "Expected min table number for index",
+	})
+	actualMinIndexTableNumber = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "cortex",
+		Name:      "actual_min_index_table_number",
+		Help:      "Actual min table number for index",
+	})
+	expectedMinChunkTableNumber = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "cortex",
+		Name:      "expected_min_chunk_table_number",
+		Help:      "Expected min table number for chunks",
+	})
+	actualMinChunkTableNumber = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "cortex",
+		Name:      "actual_min_chunk_table_number",
+		Help:      "Actual min table number for chunks",
+	})
 )
 
 func init() {
 	prometheus.MustRegister(tableCapacity)
 	syncTableDuration.Register()
+	prometheus.MustRegister(expectedMinIndexTableNumber)
+	prometheus.MustRegister(actualMinIndexTableNumber)
+	prometheus.MustRegister(expectedMinChunkTableNumber)
+	prometheus.MustRegister(actualMinChunkTableNumber)
 }
```

Comment:
Could you do an extra effort to remove this `init()`  function at all and switch to the new pattern, where we pick `registerer Prometheus.Registerer` in the `NewTableManager()` and we register metrics on that? This way we wouldn't have table manager metrics exposed by Cortex replicas not running the table manager.